src/Background/Background.vala

```diff
@@ -28,39 +28,39 @@ namespace Gala {
         public BackgroundSource background_source { get; construct; }
         public bool is_loaded { get; private set; default = false; }
         public GDesktop.BackgroundStyle style { get; construct; }
-        public string? filename { get; construct; }
+        public GLib.File? file { get; construct; }
         public Meta.Background background { get; private set; }
 
         private Animation? animation = null;
-        private Gee.HashMap<string,ulong> file_watches;
+        private Gee.HashMap<GLib.File, ulong> file_watches;
         private Cancellable cancellable;
         private uint update_animation_timeout_id = 0;
 
         private Gnome.WallClock clock;
         private ulong clock_timezone_handler = 0;
 
-        public Background (Meta.Display display, int monitor_index, string? filename,
+        public Background (Meta.Display display, int monitor_index, GLib.File? file,
                 BackgroundSource background_source, GDesktop.BackgroundStyle style) {
             Object (display: display,
                     monitor_index: monitor_index,
                     background_source: background_source,
                     style: style,
-                    filename: filename);
+                    file: file);
         }
 
         construct {
             background = new Meta.Background (display);
             background.set_data<unowned Background> ("delegate", this);
 
-            file_watches = new Gee.HashMap<string,ulong> ();
+            file_watches = new Gee.HashMap<GLib.File, ulong> ();
```

Comment:
Use `GLib.File.hash` and `GLib.File.equal` here? Gee will fallback to `direct_{hash,equal}` if the functions aren't provided here.